src/model/NPMResolver.ts

```diff
@@ -239,8 +238,59 @@ export class NPMResolver {
         return true;
     }
 
+    /**
+     * Check if the assets of a specified repository are published to the npm registry
+     */
+    public static async isRepositoryAssetsPublished(
+        repoName: string
+    ): Promise<boolean> {
+        const realRepoName = repoName === 'main' ? 'cplace' : repoName;
+        const artifactPath = `api/storage/cplace-npm-local/@cplace-assets/${realRepoName}/`;
```

Comment:
This will work as long as we have publishing of the legacy assets only for product. We need to keep in mind that this might change in the future.